src/main/webapp/app/shared/competency-selection/competency-selection.component.ts

```diff
@@ -21,10 +22,11 @@ import { finalize } from 'rxjs';
 export class CompetencySelectionComponent implements OnInit, ControlValueAccessor {
     @Input() labelName: string;
     @Input() labelTooltip: string;
-    @Input() value: any;
+    // selected competencies
+    @Input() value: Competency[] = [];
     @Input() disabled: boolean;
-    @Input() error: boolean;
-    @Input() competencies: Competency[];
+    // all course competencies
+    @Input() competencies: Competency[] = [];
```

Comment:
Tested on TS2 - I could not get any of the competencies to show up when editing the exercises. I suspect it has something to do with this line. 

The condition for loading the course competencies in `ngOnInit()` reads:

```ts
if(!this.competencies && courseId) {
  /* ... */
}
```

`!this.competencies` evaluates to `false` if `this.competencies` is an empty array, which seems to be the case when opening the edit view of an exercise.

Reply:
Yeah, you are right, I have overlooked this. 
It is fixed, please check again